src/bp/core/botpress.ts

```diff
@@ -121,14 +121,16 @@ export class Botpress {
   }
 
   async checkJwtSecret() {
-    let jwtSecret = this.config!.jwtSecret
-    if (!jwtSecret) {
-      jwtSecret = nanoid(40)
-      this.configProvider.mergeBotpressConfig({ jwtSecret })
+    // @deprecated : botpress > 11 remove jwtSecret
+    // @ts-ignore
+    let appSecret = this.config.appSecret || this.config.jwtSecret
+    if (!appSecret) {
+      appSecret = nanoid(40)
+      this.configProvider.mergeBotpressConfig({ appSecret })
       this.logger.warn(`JWT Secret isn't defined. Generating a random key...`)
     }
 
-    process.JWT_SECRET = jwtSecret
+    process.JWT_SECRET = appSecret
```

Comment:
If we are to name jwtSecret appSecret,  I guess we could do the same with the global process variable :p